django/sierra/utils/solr.py

```diff
@@ -46,21 +46,19 @@ def save(self, url=None, using='default', **kwargs):
 
 
 class Queryset(object):
-    def __init__(self, url=None, using='default', page_by=100, **kwargs):
-        if not url:
-            try:
-                url = settings.HAYSTACK_CONNECTIONS[using]['URL']
-            except KeyError:
-                raise ImproperlyConfigured('Haystack connection {} does not '
-                                           'exist.'.format(using))
-        self._conn = pysolr.Solr(url, **kwargs)
+    def __init__(self, url=None, using='default', page_by=100, conn=None,
+                 **kwargs):
+        if conn:
+            self._conn = conn
+        else:
+            self._conn = connect(url=url, using=using, **kwargs)
```

Comment:
A trick I like to use when I need a default value in case one isn't passed in is this

```python
def __init__(self, url=None, using='default', page_by=100, conn=None, **kwargs):
    self._conn = conn or connect(url=url, using=using, **kwargs)
```
The `or` will short circuit if the first value is truthy.

Either way works the same though :smile: